statement.go

```diff
@@ -284,7 +301,7 @@ func parseQueryResult(iter *spanner.RowIterator) ([]Row, []string, error) {
 			Columns: columns,
 		})
 	}
-	return rows, columnNames, nil
+	return rows, extractColumnNames(iter.Metadata.GetRowType().GetFields()), nil
```

Comment:
What is the difference between `row.ColumnNames()` and `extractColumnNames(iter.Metadata.GetRowType().GetFields())`?

Reply:
https://github.com/cloudspannerecosystem/spanner-cli/pull/191#issuecomment-2390562509

I believe there is no reason to prefer `row.ColumnNames()` rather than `.Metadata.RowType`.

* `iter.Metadata.GetRowType().GetFields()` is always available even if the result is empty or `QueryMode=PLAN`.
* `row.ColumnNames()` can't be used when the result is empty because there is no `spanner.Row` in `spanner.RowIterator`.

It means that extracting column name from `row.ColumnNames()` can't be robust.
I think we should only use `metadata.rowType`.

Reply:
Thank you for clarifying this. Yeah it makes sense to use `metadata.rowType` for getting the column information.